File: C++版/14. 最长公共前缀.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.empty())
            return "";
        sort(strs.begin(), strs.end());  //排序后，只需要比较第一个和最后一个
        int n = strs.size();
        int l = min(strs[0].size(), strs[n-1].size());
        for(int i = 0;i < l; i++){
            if(strs[0][i] != strs[n-1][i])
                return strs[0].substr(0, i);
        }
        return strs[0].substr(0, l);  //全都符合的话，最大前缀就是本身了
    }
};

File: Python版/14. 最长公共前缀.py
class Solution:
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0:
            return ""
        strs = sorted(strs)
        l = min(len(strs[0]), len(strs[len(strs)-1]))
        for i in range(l):
            if(strs[0][i] != strs[len(strs)-1][i]):
                return strs[0][0:i]
        return strs[0][0:l]